gpflow/training/optimizer.py

```diff
@@ -37,7 +37,24 @@ def make_optimize_tensor(self, model, session=None, var_list=None, **kwargs):
 
     @abc.abstractmethod
     def minimize(self, model, session=None, var_list=None, feed_dict=None,
-                 maxiter=1000, initialize=True, anchor=True, **kwargs):
+                 maxiter=1000, initialize=True, anchor=True, step_callback=None, **kwargs):
```

Comment:
Minor: Docs for `NotImplemented` method is bad idea, it does nothing. Descendants can re-define what each of the arguments means.